repo-agent/pkg/llm/gemini.go

```diff
@@ -68,6 +74,48 @@ func (g *Gemini) Setup(workspacesDir, tokensDir string) error {
 	return nil
 }
 
+func ensureSettings(geminiDir string) error {
+	settingsPath := filepath.Join(geminiDir, "settings.json")
+	if err := os.MkdirAll(geminiDir, 0755); err != nil {
+		return fmt.Errorf("failed to create .gemini directory: %v", err)
+	}
+
+	var settings map[string]interface{}
+	if _, err := os.Stat(settingsPath); err == nil {
+		data, err := os.ReadFile(settingsPath)
+		if err == nil {
+			if err := json.Unmarshal(data, &settings); err != nil {
+				log.Printf("Warning: failed to unmarshal existing settings.json: %v", err)
+			}
+		}
+	}
+
+	if settings == nil {
+		settings = make(map[string]interface{})
+	}
+
+	general, ok := settings["general"].(map[string]interface{})
+	if !ok {
+		// handle case where "general" is not a map
+		general = make(map[string]interface{})
+		settings["general"] = general
+	}
+	general["previewFeatures"] = true
+	if _, ok := settings["model"]; !ok {
+		settings["model"] = "gemini-3-pro-preview"
```

Comment:
I like this we are injecting opinionated settings. 
This will override user provided one via configdir